product_variant_configurator/models/product_configurator.py

```diff
@@ -0,0 +1,280 @@
+                res2.append(val)
+        return res2
+
+    @api.model
+    def _get_product_description(self, template, product, product_attributes):
+        name = product and product.name or template.name
+        extended = self.user_has_groups(
+            "product_variant_configurator.group_product_variant_extended_description"
+        )
+        if not product_attributes and product:
+            product_attributes = product.product_template_attribute_value_ids
+        values = self._order_attributes(template, product_attributes)
+        if extended:
+            description = "\n".join(
+                "{}: {}".format(x["value_id"].attribute_id.name, x["value_id"].name)
+                for x in values
+            )
+        else:
+            description = ", ".join([x["value_id"].name for x in values])
+        if not description:
+            return name
+        return ("%s\n%s" if extended else "%s (%s)") % (name, description)
+
+    @api.model
+    def create(self, vals):
+        """Fill `product_tmpl_id` in case `product_id` is supplied but not the
+        other one.
+        """
+        if vals.get("product_id"):
+            product = self.env["product.product"].browse(vals["product_id"])
+            if not vals.get("product_tmpl_id"):
+                vals["product_tmpl_id"] = product.product_tmpl_id.id
+            if not vals.get("product_attribute_ids"):
+                vals["product_attribute_ids"] = []
+                gen_dict = {
+                    "owner_model": self._name,
+                    "product_tmpl_id": product.product_tmpl_id.id,
+                }
+                for att_val in product._get_product_attributes_values_dict():
+                    att_val.update(gen_dict)
+                    vals["product_attribute_ids"].append((0, 0, att_val))
+        return super().create(vals)
+
+    def unlink(self):
+        """Mimic `ondelete="cascade"`."""
+        attributes = self.mapped("product_attribute_ids")
+        result = super(ProductConfigurator, self).unlink()
+        if result:
+            attributes.unlink()
+        return result
+
+    def create_variant_if_needed(self):
+        """ Create the product variant if needed.
+
+        It searches for an existing product with the selected attributes. If
+        not found, create a new product.
+
+        :returns: the product (found or newly created)
+        """
+        self.ensure_one()
+        if self.product_id:
+            return self.product_id
+        product_obj = self.env["product.product"]
+        product = product_obj._product_find(
+            self.product_tmpl_id, self.product_attribute_ids,
+        )
+        if not product:
+            product = product_obj.create(
+                {
+                    "product_tmpl_id": self.product_tmpl_id.id,
+                    "product_template_attribute_value_ids": [
+                        (6, 0, self.product_attribute_ids.mapped("value_id").ids)
```

Comment:
`self.product_attribute_ids.mapped("value_id").ids` will return `product.attribute.value` ids. But `product_template_attribute_value_ids` is a `product.template.attribute.value` field.

We need to build a `product.template.attribute.value` from `self.product_attribute_ids.mapped("value_id")`. I've done a PR with my fix on your fork [here](https://github.com/ForgeFlow/product-variant/pull/1).

WYT?

Reply:
Looks great!